frontend/packages/topology/src/behavior/useDndDrag.tsx

```diff
@@ -131,10 +130,13 @@ export const useDndDrag = <
     React.useCallback(
       (node: SVGElement | null) => {
         if (node) {
-          let operationChangeEvents: {
-            begin: [number, number, number, number];
-            drag: [number, number, number, number];
-          } | null = null;
+          let operationChangeEvents:
+            | {
+                begin: [number, number, number, number];
+                drag: [number, number, number, number];
+              }
+            | undefined;
+          let operation: DragSpecOperation | undefined;
```

Comment:
`| undefined` isn't needed for typescript types. If you're dynamically typing a variable, undefined is inherited, lack of being equal to something is undefined. If you're creating a stand alone type, there is `?` to indicate optionality.

Reply:
The typescript rules in the `topology` project are very strict (`strictNullChecks = true`). If you do not define the type to be allowed a value of `undefined` then you cannot assign it `null` or `undefined`.

Take your pick:

1. Automatically initialized to `undefined`:
`let foo: <type> | undefined;`
Then you can do:
`foo = undefined`

or

2. Requires assignment to null:
`let foo: <type> | null = null;`
Then you can do:
`foo = null`

Reply:
Ah, neat. Guess I never worked with `strictNullChecks` before. Maybe it didn't exist back in the day 🤷‍♂ ... either way, cool. In favour of no 2. JavaScript's `undefined` should be kept away from variables imo, and only used in rare cases, like checking the existence of a property.

Reply:
Ah yeah, stopped using TS just before 2.0 rolled out.

> --strictNullChecks switches to a new strict null checking mode [in TypeScript 2.0].

Reply:
So `let operation: DragSpecOperation | undefined;` is `undefined` instead of `null` because further down we assign a value to `operation` from `specRef.current.operation` which can be `undefined`.

But `operationChangeEvents` makes sense to be `| null`.

`strictNullChecks` gives me headaches.